python/ray/serve/_private/http_state.py

```diff
@@ -54,6 +62,14 @@ def __init__(
             actor_name=self._actor_name,
             status=self._status,
         )
+        self._long_poll_client = LongPollClient(
+            ray.get_actor(controller_name, namespace=SERVE_NAMESPACE),
+            {
+                LongPollNamespace.ACTIVE_NODES: self._update_draining,
+            },
+            call_in_event_loop=get_or_create_event_loop(),
+        )
```

Comment:
I don't think we should be using the long poll client for this because it means that the controller is calling into itself for updates. This could have some strange side effects (I'm not sure if an actor calling into itself is well-tested). For example, I could imagine an edge case leading to deadlock where the loop is blocked calling into the controller, not allowing itself to respond to the actor method call.

Any reason not to put this in `update()` as I suggested previously? Should be straightforward, just return the map from `_update_active_nodes()` and pass it in.

Reply:
ooo I think I misunderstood when and how to use the long poll. I thought by putting long poll client on both http proxy state and http proxy, it would ensure the consistency between the two, and also save on the communication, since the long poll host just need to pass the message once and both will get updated.  

Will keep the long poll client on the http proxy, but drop this one on the http proxy state and just use `update()` to update http proxy state.